utils/advanced_stats_lib.py

```
@@ -0,0 +1,201 @@
+        model_type (str): Model family influencing selection logic (e.g., "logistic", "linear", "cox").
+    
+    Returns:
+        dict: A dictionary with keys:
+            "method" (str): The selected CI method ("wald" or "profile", or the provided method if not "auto").
+            "note" (str): Informational note about auto-selection when applicable, otherwise an empty string.
+    
+    Raises:
+        ValueError: If `method` is not one of "auto", "wald", or "profile".
+    """
+    if method not in {"auto", "wald", "profile"}:
+        raise ValueError(f"Unsupported CI method: {method}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)